lib/googleauth/credentials.rb

```diff
@@ -0,0 +1,156 @@
+# Copyright 2017, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+require 'forwardable'
+require 'json'
+require 'signet/oauth_2/client'
+
+require 'googleauth/default_credentials'
+
+module Google
+  module Auth
+    # This class is intended to be inherited by API-specific classes
+    # which overrides the SCOPE constant.
+    class Credentials
+      TOKEN_CREDENTIAL_URI = 'https://accounts.google.com/o/oauth2/token'.freeze
+      AUDIENCE = 'https://accounts.google.com/o/oauth2/token'.freeze
+      SCOPE = [].freeze
+      PATH_ENV_VARS = [].freeze
```

Comment:
I'm not clear why the defaults were removed here. Some of the existing subclasses of `Google::Cloud::Credentials` may expect to append their vars to these base sets.

Reply:
The defaults are not listed because the defaults listed [in google-cloud-ruby](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-core/lib/google/cloud/credentials.rb#L32) are specific to google cloud. 

For the cloud case, we can either append those default locations to the extended class, or we can have Google::Cloud::Credentials extend this class with those locations.

Reply:
Okay. We'll just have Google::Cloud::Credentials set them specially when we move it over to extend this class.

Reply:
Ack.